service/matching/matcher.go

```diff
@@ -646,8 +646,8 @@ func (tm *TaskMatcher) unregisterBacklogTask(task *internalTask) {
 // getBacklogAge is the latest age across all backlogs re-directing to this matcher; may momentarily
 // be 0 cause of race conditions when no reader pushes a task into the matcher at this moment
 func (tm *TaskMatcher) getBacklogAge() time.Duration {
-	tm.backlogTasksLock.Lock()
-	defer tm.backlogTasksLock.Unlock()
+	tm.backlogTasksLock.RLock()
+	defer tm.backlogTasksLock.RUnlock()
```

Comment:
what's the need for RW lock? generally, we should prefer simple locks unless they are not enough.

Reply:
Simple locks in this case could be *okay* too but I changed this to using a RW mutex because the advent of a busy loop (or eventually) which will have multiple successive reads made me ponder if I should really be locking/unlocking one lock.

I'm okay to change this back to the way it was (one lock) and have done so for now